File: modules/hero-carousel/android/src/main/java/expo/modules/herocarousel/HeroCarouselModule.kt
package expo.modules.herocarousel

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class HeroCarouselModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("HeroCarousel")

    View(HeroCarouselExpoView::class) {
      Events("onItemPress", "onFilterToggle")

      // Everything the view renders arrives as one JSON string: the slides,
      // the image auth headers and the filter menu. The string prop exists
      // for iOS's sake (typed props are applied there with `try? prop.set`,
      // which swallows a failed conversion), and it is the reason this view
      // could be added under the same name with no JS changes at all.
      Prop("payload") { view: HeroCarouselExpoView, payload: String ->
        view.setPayload(payload)
      }
    }
  }
}

File: modules/hero-carousel/ios/HeroCarouselModule.swift
import ExpoModulesCore

public class HeroCarouselModule: Module {
  public func definition() -> ModuleDefinition {
    Name("HeroCarousel")

    View(HeroCarouselExpoView.self) {
      Events("onItemPress", "onFilterToggle")

      // Everything the view renders arrives as one JSON string: the slides
      // and the image auth headers. Typed Record/Dictionary props proved
      // unreliable here — expo-modules-core applies view props with
      // `try? prop.set(...)` (ExpoFabricView.updateProps), so a conversion
      // that fails is swallowed silently and the view just renders empty
      // with no error anywhere. A String prop has no such failure mode, and
      // the payload stays portable for a future Android implementation.
      Prop("payload") { (view: HeroCarouselExpoView, payload: String) in
        view.setPayload(payload)
      }
    }
  }
}
